### PR TITLE
[Peer Review] Add Windows artifact for merges to main

### PR DESCRIPTION
**Pull request by SparkingSpork**
### What does this Pull Request accomplish?
To facilitate releases, this PR adds Windows artifacts containing the server binary any time a merge to main occurs.

### Why should this Pull Request be merged?
To allow usage of the Windows artifacts for incremental usage and creating releases without requiring a local build.

### What testing has been done?
Ran a private workflow to ensure the artifact contains the right files. See [here](https://github.com/ni/grpc-device/actions/runs/808000693).

### CODE REVIEW

**Review by kreimnitz — APPROVED** [review]

**Review by reckenro — APPROVED** [review]
This looks good for the main branch. The one thing I'm not sure about is how to have these artifacts for the [releases/1.0 branch](https://github.com/ni/grpc-device/tree/releases/1.0). The main branch has some ni-sync stuff in it now that we don't want in a 1.0.1 release so we made the releases/1.0 branch to cherry pick bug fixes we want in the 1.0.x releases.

This goes for both the linux and windows artifacts but maybe in that branch it makes sense to submit a change so the if statement refers to that branch instead of main. @jonlugoatx , thoughts on this?

**Comment by SparkingSpork** [comment]
@reckenro, @jonlugoatx  - We would simply need to add pushes to the `release/**` branches and a similar check for the specific actions to cover release branches. I can make that change in this PR if desired.

File: .github/workflows/build_cmake.yml [hunk 1]
```diff
@@ -123,3 +123,13 @@ jobs:
         name: ni-grpc-device-server-linux-glibc2_23
         path: ni-grpc-device-server-linux-glibc2_23.tar
         retention-days: 5
+
+    - name: Upload Windows Server Binaries Artifact
+      uses: actions/upload-artifact@v2
+      if: "(github.ref == 'refs/heads/main') && (runner.os == 'Windows')"
```
Comment (jonlugoatx) [inline .github/workflows/build_cmake.yml:129]: This should also run on `refs/heads/releases/*`

**Review by jonlugoatx — APPROVED** [review]

File: .github/workflows/build_cmake.yml (hunk 1 above)
Comment (SparkingSpork) [reply .github/workflows/build_cmake.yml]: I used `startsWith` as it wasn't clear from the documentation if operators accepted wildcard filters.

**Review by reckenro — APPROVED** [review]

**Pull request merged**